File: dailyques/Q3047.py
#Q3047.py
class Solution:
    def largestSquareArea(self, bl, tr):
        n = len(bl)
        ans = 0
        for i in range(n):
            for j in range(i+1, n):
                x1 = max(bl[i][0], bl[j][0])
                y1 = max(bl[i][1], bl[j][1])
                x2 = min(tr[i][0], tr[j][0])
                y2 = min(tr[i][1], tr[j][1])
                if x2 > x1 and y2 > y1:
                    side = min(x2-x1, y2-y1)
                    ans = max(ans, side*side)
        return ans

File: dailyques/Q3047.cpp
//Q3047.cpp
class Solution {
public:
    long long largestSquareArea(vector<vector<int>>& bl, vector<vector<int>>& tr) {
        long long ans = 0;
        int n = bl.size();
        for(int i=0;i<n;i++){
            for(int j=i+1;j<n;j++){
                long long x1 = max(bl[i][0], bl[j][0]);
                long long y1 = max(bl[i][1], bl[j][1]);
                long long x2 = min(tr[i][0], tr[j][0]);
                long long y2 = min(tr[i][1], tr[j][1]);
                if(x2 > x1 && y2 > y1){
                    long long side = min(x2-x1, y2-y1);
                    ans = max(ans, side*side);
                }
            }
        }
        return ans;
    }
};
